src/org/opensolaris/opengrok/util/OptionParser.java

```diff
@@ -0,0 +1,701 @@
+     * is indicated by surrounding the expression with '/' character. For
+     * example, "/pattern/" indicates that the only value acceptable is the
+     * word 'pattern'.
+     * 
+     * Any string that does not start with a '-', '=', or '/' is used as a
+     * description for the option in the summary. Multiple descriptions may
+     * be given; they will be shown on additional lines.
+     * 
+     * @return Option
+     */
+    public Option on(Object ... args) {
+
+        Option opt = new Option();
+        
+        for (Object arg : args) {
+            if (arg instanceof String) {
+                String argument = (String) arg;
+                if (argument.startsWith("--")) {
+                    // handle --switch --switch=ARG --switch=[OPT] --switch PLACE
+                    String[] parts = argument.split("[ =]");
+                    
+                    if (parts.length == 1) {
+                        opt.addAlias(parts[0]);
+                    } else {
+                        opt.set(parts[0], parts[1]);
+                    }
+                } else if (argument.startsWith("-")) {
+                    // handle -x -xARG -x=ARG -x[OPT] -x[=OPT] -x PLACE
+                    String optName = argument.substring(0,2);
+                    String remainder = argument.substring(2);
+                    opt.set(optName, remainder);
+                    
+                } else if (argument.startsWith("=")) {
+                    opt.setArgument(argument.substring(1));
+                } else if (argument.startsWith("/")) {
+                    // regular expression (sans '/'s)
+                    opt.setPattern(argument.substring(1,argument.length()-1));
+                } else {
+                    // this is description
+                    opt.addDescription(argument);
+                }
+            // This is indicator for a set of specific allowable option values
+            } else if (arg instanceof String[]) {
+                opt.setAllowedValues((String[])arg);
+            // This is indicator for option value data type 
+            // to which the parser will take and convert.
+            } else if (arg instanceof Class) {
+                opt.setValueType((Class<?>)arg);
+            }
+        }
+        optionList.add(opt);
+        usageSummary.add(opt);
+        return opt;
+    }
+    
+    private String argValue(String arg, Boolean mandatory) {
+        // Initially assume that the given argument is going
+        // to be the option's value. Note that if the argument
+        // is actually another option (starts with '-') then
+        // there is no value available. If the option is required
+        // to have a value, null is returned. If the option
+        // does not require a value, an empty string is returned.
+        String value = arg;
+        Boolean isOption = value.startsWith("-");
+        
+        if (mandatory) {
+            if(isOption ) {
```

Comment:
style

Reply:
by style do you mean I missed a space between 'if' and '(' ?

Reply:
yep + missing space between `if` and left paren